services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/tormaystarkastelu/LuokitteluRajaArvotService.kt

```diff
@@ -1,47 +1,31 @@
 package fi.hel.haitaton.hanke.tormaystarkastelu
 
-data class RajaArvo(val luokittelu: Luokittelu, val minimumValue: Int)
-
-interface LuokitteluRajaArvotService {
-
-    fun getHaittaAjanKestoLuokka(days: Int): Int
-    fun getLiikennemaaraLuokka(volume: Int): Int
-    fun getBussiLiikenneRuuhkaLuokka(countOfRushHourBuses: Int): Int
-
-}
-
-class LuokitteluRajaArvotServiceHardCoded : LuokitteluRajaArvotService {
-
-    private val rajaArvotHaittaAjanKesto = listOf(
-            RajaArvo(HaittaAjanKestoLuokittelu.YLI_3KK, 91),
-            RajaArvo(HaittaAjanKestoLuokittelu.ALLE_3KK_YLI_2VK, 14),
-            RajaArvo(HaittaAjanKestoLuokittelu.ALLE_2VK, 0)
-    )
-
-    private val rajaArvotLiikennemaara = listOf(
-            RajaArvo(LiikenneMaaraLuokittelu.YLI_10K,10000),
-            RajaArvo(LiikenneMaaraLuokittelu._5K_10K, 5000),
-            RajaArvo(LiikenneMaaraLuokittelu._1500_5K, 1500),
-            RajaArvo(LiikenneMaaraLuokittelu._500_1500, 500),
-            RajaArvo(LiikenneMaaraLuokittelu.ALLE_500, 1),
-            RajaArvo(LiikenneMaaraLuokittelu.EI_LIIKENNETTA, 0)
-    )
-
-    private val rajaArvotBussi = listOf(
-            RajaArvo(BussiLiikenneLuokittelu.KAMPPI_RAUTATIENTORI, 21),
-            RajaArvo(BussiLiikenneLuokittelu.RUNKOLINJA, 11),
-            RajaArvo(BussiLiikenneLuokittelu.RUNKOLINJAMAINEN, 5),
-            RajaArvo(BussiLiikenneLuokittelu.PERUS, 0),
-            RajaArvo(BussiLiikenneLuokittelu.EI_VAIKUTA, -1)
-    )
-
-    override fun getHaittaAjanKestoLuokka(days: Int) =
-            rajaArvotHaittaAjanKesto.find { it.minimumValue <= days }?.luokittelu?.value ?: 0
-
-    override fun getLiikennemaaraLuokka(volume: Int) =
-            rajaArvotLiikennemaara.find { it.minimumValue <= volume }?.luokittelu?.value ?: 0
-
-    override fun getBussiLiikenneRuuhkaLuokka(countOfRushHourBuses: Int) =
-            rajaArvotBussi.find { it.minimumValue <= countOfRushHourBuses }?.luokittelu?.value ?: 0
-
+object LuokitteluRajaArvotService {
+
+    fun getHaittaAjanKestoLuokka(days: Int): Int =
+        when {
+            days >= 91 -> HaittaAjanKestoLuokittelu.YLI_3KK.value
+            days >= 14 -> HaittaAjanKestoLuokittelu.ALLE_3KK_YLI_2VK.value
+            days >= 0 -> HaittaAjanKestoLuokittelu.ALLE_2VK.value
+            else -> 0
```

Comment:
It seems that the only way to end up with zero is a negative number as an argument. Would it make sense to somehow validate the input in these functions (or is that done somewhere else)?

Reply:
This was how the previous implementation worked. Refactoring shouldn't change the behaviour.

The only way to get a negative input here is if the hanke end date is before the start date, which is validated to never happen.